### PR TITLE
Part4

### PR DESCRIPTION
**Pull request by remingtonchan**
Implemented pow functions for each of the UDTs.
Implemented constructors for Point that accepts UDTs as well as uses delegate constructors.
Fixed warning by passing by reference inside multiply functions instead of pass by value.
Moved out definitions to out of class (except operator)

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -48,103 +48,27 @@ Project 4: Part 4 / 9
...
-    //                          Point tests
-    // ------------------------------------------------------------
-    FloatType ft2(3.0f);
-    DoubleType dt2(4.0);
-    IntType it2(5);
-    float floatMul = 6.0f;
+    Point(float a, float b);
+    Point(FloatType& a, FloatType& b);
+    Point(DoubleType& a, DoubleType& b);
+    Point(IntType& a, IntType& b);
```
Comment (matkatmusic) [inline main.cpp:60]: If you don't need to modify the function argument, pass by `const reference` for non-trivial types, pass by value for trivial types (C++ primitives mostly)
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -48,103 +48,27 @@ Project 4: Part 4 / 9
...
-    Point p1(ft2, ft2);
-    p1.toString();   
-    std::cout << "Multiplication factor: " << ft2 << std::endl;
-    p1.multiply(ft2); 
-    p1.toString();   
-    std::cout << "---------------------\n" << std::endl;
+    Point& multiply(float m);
+    Point& multiply(FloatType& m);
+    Point& multiply(DoubleType& m);
+    Point& multiply(IntType& m);
```
Comment (matkatmusic) [inline main.cpp:65]: pass by const reference
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
The expected output is not being produced.  
Please fix your `Point::toString()` function as well as correct the things marked below:

File: main.cpp (hunk 1 above)
Comment (remingtonchan) [reply main.cpp:60]: I'm stuck here with ambiguous constructors when I turn the arguments into `const reference` vs just `reference`

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**